Firestore/core/test/firebase/firestore/local/lru_garbage_collector_test.cc

```diff
@@ -0,0 +1,740 @@
+  QueryData query_data = NextTestQuery();
+  query_cache_->AddTarget(query_data);
+  return query_data;
+}
+
+void LruGarbageCollectorTest::UpdateTargetInTransaction(
+    const QueryData& query_data) {
+  QueryData updated =
+      query_data.WithSequenceNumber(persistence_->current_sequence_number());
+  query_cache_->UpdateTarget(updated);
+}
+
+DocumentKey LruGarbageCollectorTest::CreateDocumentEligibleForGc() {
+  DocumentKey key = NextTestDocKey();
+  MarkDocumentEligibleForGc(key);
+  return key;
+}
+
+DocumentKey
+LruGarbageCollectorTest::CreateDocumentEligibleForGcInTransaction() {
+  DocumentKey key = NextTestDocKey();
+  MarkDocumentEligibleForGcInTransaction(key);
+  return key;
+}
+
+void LruGarbageCollectorTest::MarkDocumentEligibleForGc(
+    const DocumentKey& doc_key) {
+  persistence_->Run("Removing mutation reference",
+                    [&] { MarkDocumentEligibleForGcInTransaction(doc_key); });
+}
+
+void LruGarbageCollectorTest::MarkDocumentEligibleForGcInTransaction(
+    const DocumentKey& doc_key) {
+  persistence_->reference_delegate()->RemoveMutationReference(doc_key);
+}
+
+void LruGarbageCollectorTest::AddDocument(const DocumentKey& doc_key,
+                                          TargetId target_id) {
+  query_cache_->AddMatchingKeys(DocumentKeySet{doc_key}, target_id);
+}
+
+void LruGarbageCollectorTest::RemoveDocument(const DocumentKey& doc_key,
+                                             TargetId target_id) {
+  query_cache_->RemoveMatchingKeys(DocumentKeySet{doc_key}, target_id);
+}
+
+Document LruGarbageCollectorTest::CacheADocumentInTransaction() {
+  Document doc = NextTestDocument();
+  document_cache_->Add(doc);
+  return doc;
+}
+
+SetMutation LruGarbageCollectorTest::MutationForDocument(
+    const DocumentKey& doc_key) {
+  return SetMutation(doc_key, test_value_, Precondition::None());
+}
+
+DocumentKey LruGarbageCollectorTest::NextTestDocKey() {
+  return Key("docs/doc_" + std::to_string(++previous_doc_num_));
+}
+
+Document LruGarbageCollectorTest::NextTestDocumentWithValue(ObjectValue value) {
+  DocumentKey key = NextTestDocKey();
+  return Document(std::move(value), std::move(key), Version(2),
+                  DocumentState::kSynced);
+}
+
+Document LruGarbageCollectorTest::NextTestDocument() {
+  return NextTestDocumentWithValue(test_value_);
+}
+
+#pragma mark - tests
+
+TEST_P(LruGarbageCollectorTest, PickSequenceNumberPercentile) {
+  const int num_test_cases = 5;
+  struct Case {
+    // number of queries to cache
+    int queries;
+    // number expected to be calculated as 10%
+    int expected;
+  };
+  struct Case test_cases[num_test_cases] = {
+      {0, 0}, {10, 1}, {9, 0}, {50, 5}, {49, 4}};
+
+  for (int i = 0; i < num_test_cases; i++) {
+    // Fill the query cache.
+    int num_queries = test_cases[i].queries;
+    int expected_tenth_percentile = test_cases[i].expected;
+    NewTestResources();
+    for (int j = 0; j < num_queries; j++) {
+      AddNextQuery();
+    }
```

Comment:
Nit: many of the functions in this file would benefit from some blank lines. Can you add an empty line here, right after the loop?